### PR TITLE
Introduce UpdateVirtualProductFileCommand

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Introduce UpdateVirtualProductFileCommand. It updates VirtualProductFile (formerly known as ProductDownload) enity & allows replacing its file. This Pr includes a little feature update - we do not provide ability to delete the file separately from entity, but instead we allow using the replace behavior. So if you want to delete the file you have to delete whole ProductDownload entity, or you can update it and replace current file with another one. Add behat test to cover this command
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | part of #14772
| How to test?      | CI :heavy_check_mark: 
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23386)

### CODE REVIEW

**Comment by zuk3975** [comment]
todo: missing behat tests

**Review by jolelievre — DISMISSED** [review]
Missing behat tests but all the rest is ok for me, so I give an early approval before my holidays 😉

File: src/Adapter/Product/VirtualProduct/CommandHandler/UpdateVirtualProductFileHandler.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        $this->virtualProductUpdater = $virtualProductUpdater;
+        $this->virtualProductFileRepository = $virtualProductFileRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateVirtualProductFileCommand $command): void
+    {
+        $virtualProductFile = $this->virtualProductFileRepository->get($command->getVirtualProductFileId());
```
Comment (sowbiba) [inline src/Adapter/Product/VirtualProduct/CommandHandler/UpdateVirtualProductFileHandler.php:69]: Don't we need to test that $virtualProductFile is not null ?

**Review by sowbiba — APPROVED** [review]
Nice job. Just a question

File: src/Adapter/Product/VirtualProduct/CommandHandler/UpdateVirtualProductFileHandler.php (hunk 1 above)
Comment (zuk3975) [reply src/Adapter/Product/VirtualProduct/CommandHandler/UpdateVirtualProductFileHandler.php:69]: no because Updater accepts null|string as filepath

Comment (zuk3975) [reply src/Adapter/Product/VirtualProduct/CommandHandler/UpdateVirtualProductFileHandler.php:69]: nope, null check is handled in repository. It gets the entity or throws exception

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @zuk3975
